ui/src/actions/user.js

```diff
@@ -88,6 +88,7 @@ export const changeUsername = (newUsername) => async (dispatch, getState) => {
         msg: "Your username has now been updated.",
       })
     );
+    window.location.reload();
```

Comment:
Not a fan of this. We should update our state without having to refresh the whole page.

Reply:
I have zero frontend experience, this was the closest thing that worked for me. 
If you want, I can leave this out, and you can fix it separately (or just push to this branch), or if you'd like to point me in the right direction, I could try to figure it out

Reply:
Let's omit this unless you want to give it a try. I've been meaning to convert the sidebar to use RTKQ anyway which would mean you'd have to basically call a `refetch` function.

Reply:
I attempted to fix this, take a look and let me know if you like the other implementation.  If not, I'll remove it and let someone else take care of it.